lib/signaling/v2/peerconnection.js

```diff
@@ -488,10 +502,46 @@ class PeerConnectionV2 extends StateMachine {
       this._log.info('ICE reconnected');
     }
 
+    if (iceConnectionState === 'connected') {
+      this._mediaMonitor.start(() => {
+        this._log.debug(`makarand: Media Monitor detected inactivity while in "${this._lastIceConnectionState}" state`);
+        if (this._lastIceConnectionState === 'disconnected') {
+          this._mediaMonitor.stop();
+          if (!this._shouldRestartIce && !this._isRestartingIce) {
+            this._shouldRestartIce = true;
+            this.offer();
+          }
+        }
+      });
+    } else if (iceConnectionState === 'failed' || iceConnectionState === 'closed') {
```

Comment:
@makarandp0 I don't think this will be necessary with the new way of checking inactivity, since we are using `setTimeout` instead of `setInterval`.

Reply:
updated this now that we've decided to listen only after disconnect.